libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -665,14 +665,46 @@ std::shared_ptr<arrow::Schema> make_experimental_schema(const type& t) {
   std::vector<std::shared_ptr<arrow::Field>> arrow_fields;
   arrow_fields.reserve(rt.num_leaves());
   for (const auto& [field, index] : rt.leaves()) {
-    auto field_ptr
-      = arrow::field(rt.key(index), make_experimental_type(field.type));
-    arrow_fields.emplace_back(std::move(field_ptr));
+    arrow_fields.emplace_back(
+      make_experimental_field(rt.key(index), field.type));
   }
   auto metadata = arrow::key_value_metadata({{"name", std::string{t.name()}}});
   return std::make_shared<arrow::Schema>(arrow_fields, metadata);
 }
 
+std::shared_ptr<arrow::Field>
+make_experimental_field(const std::string& name, const type& t) {
```

Comment:
This should work on a `const record_type::field_view& field`  rather than the two listed parameters.

Reply:
I think that does not currently work, because we're calling it like this:
```c++
  for (const auto& [field, index] : rt.leaves()) {
    arrow_fields.emplace_back(
      make_experimental_field(rt.key(index), field.type));
  }
```

IIUC, the `rt.key(index)` is necessary because we're building the fully qualified name to the leaf field, and `field.name` would not suffice.

I _think_ this should be possible as soon as we properly represent the VAST type in the Arrow schema in terms of the full tree, not a flattened view of leaves attached directly to the root.

Reply:
One thing you could do is to call `auto frt = flatten(rt)` once and then iterate over its `frt.fields()`. This would express the flattening more clearly as well.

Reply:
I'll give that a try. I like the cleaner duality between `make_vast_type` and `make_experimental_field`